lib/model/folder_sendrecv.go

```diff
@@ -311,6 +350,12 @@ func (f *sendReceiveFolder) pullerIteration(ignores *ignore.Matcher, ignoresChan
 		return true
 	})
 
+	select {
+	case <-f.ctx.Done():
+		return -1, nil, nil
```

Comment:
Not sure why the `-1, nil, nil` return in these cases. Probably it doesn't matter because whoever is looking at the return is also going to exit, but if we've done 39 changes already I see no reason to lie about it?

Reply:
`-1, nil, nil` was just my way of returning an error - returning `changed, fileDeletions, dirDeletions` feels like lying too (we didn't necessarily change all of them). I think the clean thing to do here is add an error return value.